metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py

```diff
@@ -71,93 +71,259 @@
+
+    The databricks-sql thrift_http_client requires thrift.transport.THttpClient.THttpClient
+    to be accessible. This is achieved by importing the required modules in the right order.
+    """
+    try:
+        # Import thrift submodules - this makes them accessible as attributes
+        import thrift.transport.THttpClient  # noqa: F401 # Used to make thrift.transport accessible
+
+        logger.debug("Successfully imported required thrift modules")
+    except Exception as e:
+        logger.debug(f"Could not import thrift modules: {e}")
+        raise
+
+
+def _mask_proxy_credentials(url: Optional[str]) -> str:
+    """Mask credentials in proxy URL for safe logging."""
+    if not url:
+        return "None"
+
+    try:
+        parsed = urllib.parse.urlparse(url)
+        if parsed.username:
+            # Replace credentials with masked version
+            masked_netloc = parsed.netloc
+            if parsed.username and parsed.password:
+                masked_netloc = masked_netloc.replace(
+                    f"{parsed.username}:{parsed.password}@", f"{parsed.username}:***@"
+                )
+            elif parsed.username:
+                masked_netloc = masked_netloc.replace(
+                    f"{parsed.username}@", f"{parsed.username}:***@"
+                )
+
+            return urllib.parse.urlunparse(
+                (
+                    parsed.scheme,
+                    masked_netloc,
+                    parsed.path,
+                    parsed.params,
+                    parsed.query,
+                    parsed.fragment,
+                )
+            )
+        else:
+            return url
+    except Exception:
+        return "***INVALID_URL***"
+
+
 # Apply databricks-sql < 3.0 proxy authentication fix at module import time
 # This implements the same fix as Databricks PR #354 to resolve
 # "407 Proxy Authentication Required" errors that occur even when
 # all proxy environment variables are correctly set.
 # https://github.com/databricks/databricks-sql-python/pull/354
+#
+# Note: This fix may not work with all thrift versions due to compatibility issues
+# between databricks-sql-connector 2.9.6 and newer thrift versions. The fix will
+# gracefully fail with a warning if thrift compatibility issues are detected.
+# The main SQL functionality will continue to work normally without this fix.
+def _log_proxy_environment():
+    """Log detected proxy environment variables for debugging."""
+    proxy_env_vars = {}
+    for var in PROXY_VARS:
+        value = os.environ.get(var)
+        if value:
+            masked_value = _mask_proxy_credentials(value)
+            proxy_env_vars[var] = masked_value
+
+    if proxy_env_vars:
+        logger.info(f"Detected proxy environment variables: {proxy_env_vars}")
+    else:
+        logger.debug("No proxy environment variables detected")
+
+
+def _create_patched_open_method(original_open):
```

Comment:
May we move the patch related stuff to a eg `proxy_patch.py` file?
I've seen that pattern in other cases and while we may not be consistent with that, I thiink it's good